scripts/validate_mixer.py

```diff
@@ -0,0 +1,935 @@
+import yaml
+import json
+import sys
+import jq
+from jsonpath_ng.ext import parse as parse_jsonpath
+import re
+import boto3
+import random
+import tempfile
+import os
+import shutil
+from botocore.exceptions import ClientError
+import json
+import gzip
+import io
+import itertools
+from collections import defaultdict
+import subprocess
+import signal
+import smart_open
+from tqdm import tqdm
+import importlib
+import traceback
+import shutil
+
+s3_client = boto3.client('s3')
+
+def load_config(config_path):
+    """Load the configuration file (YAML or JSON)."""
+    try:
+        with open(config_path, 'r') as file:
+            if config_path.endswith('.yaml') or config_path.endswith('.yml'):
+                return yaml.safe_load(file)
+            elif config_path.endswith('.json'):
+                return json.load(file)
+            else:
+                raise ValueError("Unsupported file format. Use .yaml, .yml, or .json")
+    except Exception as e:
+        print(f"Error loading config file: {str(e)}")
+        sys.exit(1)
+
+def keyboard_interrupt_handler(signal, frame):
+    print("\n\nScript interrupted by user. Exiting ...")
+    sys.exit(0)
+
+# Register the custom handler
+signal.signal(signal.SIGINT, keyboard_interrupt_handler)
+
+def extract_attribute_names_from_filters(filters):
+    attribute_names = set()
+    for filter_expr in filters:
+        # Extract attribute names from JSONPath expressions
+        matches = re.findall(r'@\.([a-zA-Z0-9_]+)', filter_expr)
+        attribute_names.update(matches)
+    return attribute_names
+
+def validate_config_structure(config):
+    """Validate the basic structure of the configuration."""
+    required_fields = ['streams', 'processes']
+    errors = []
+
+    for field in required_fields:
+        if field not in config:
+            errors.append(f"Missing required field: {field}")
+
+    if 'streams' in config:
+        if not isinstance(config['streams'], list):
+            errors.append("'streams' should be a list")
+        else:
+            for i, stream in enumerate(config['streams']):
+                stream_errors = validate_stream(stream, i)
+                errors.extend(stream_errors)
+
+    if 'processes' in config and not isinstance(config['processes'], int):
+        errors.append("'processes' should be an integer")
+
+    return errors
+
+def validate_stream(stream, index):
+    """Validate an individual stream configuration."""
+    required_fields = ['name', 'documents', 'attributes', 'output']
+    errors = []
+
+    for field in required_fields:
+        if field not in stream:
+            errors.append(f"Stream {index}: Missing required field: {field}")
+
+    if 'documents' in stream and not isinstance(stream['documents'], list):
```

Comment:
Worth making a helper function to reduce a bit of redundant code, like check_type_if_present(structure,type,errors)

Reply:
Another option would be to create a pydantic class or other validation config for each structure being validated, but I think with the fairly limited number of times you're currently using this pattern, it might be a little overkill.

Reply:
I added "check_type_if_present" function, now it lives in config_handler.py line 71